Add CLI Tool and updated README for Concerto Conformance Suite (#19)
accordproject/concerto-conformance: README.md

# Concerto Conformance Test Suite
## Desription

> The Concerto Conformance Test Suite provides a standardized, automated testing suite to validate models and semantic behavior across Accord Project Concerto implementations.

Reviewer: ```suggestion
The Concerto Conformance Test Suite provides a standardized, automated testing suite to validate models and semantic behavior across Accord Project's Concerto implementation.
```

> ## Overview

Reviewer: You should split the headers for readability.
```suggestion

## Overview
```

This repository includes:

> 1. A curated set of semantic validation rules

Reviewer: Is this a "curated" sample? I think it's all the semantic rules we've discovered so far isn't it?

2. Comprehensive valid and invalid model examples
3. Tests written using Cucumber, offering behavior-driven, human-readable test definitions

> 4. Support for both JavaScript and C# environments

Reviewer: ```suggestion
4. Support for both JavaScript and C# runtimes
```

The suite specifically tests core components of the Concerto ecosystem:

> 1. The Parser from `@accordproject/concerto-cto`

Reviewer: We don't specifically test the parser, that's more of a side effect.

> 2. The ModelFile and ModelManager from `@accordproject/concerto-core`

Reviewer: ```suggestion
2. The `ModelFile` and `ModelManager` classes from `@accordproject/concerto-core`
```

## Getting started
1. Install dependencies:
    `npm install`
2. Run the Javascript test suite:
    `npm test`
3. Run tests for C#:
    `npm run test:csharp`
## Interactive CLI:
You can also use the built-in CLI for a guided setup:
    `npm start`   
The CLI allows you to:   
1. Provide custom ModelManager, Parser, or ModelFile sources for testing
2. Choose between JavaScript or C# environments
3. Automatically generate test files with proper imports
4. Run the relevant conformance tests seamlessly

> ## Future Roadmap

Reviewer: I think you can add those to the project board. You don't need to add them here unless you don't have time to add these before GSoC finishes. But keeping them is fine as well. Just keep in mind to have them up to date.
